project: saving history docs (#1067)
mia-platform/documentation: docs/development_suite/set-up-infrastructure/enhanced-project-workflow.md

- `Branches` and `Tags` become **`Revisions`** and **`Versions`**
- the `Commit history` becomes **`Revision history`**

Developers can manage *revisions* and *versions* from the dedicated sections inside the Overview area. Revisions can be accessed and deleted from the Revisions management page:

![Revisions management page](img/revisions-overview.png)

> Versions, a feature similar to Git _tags_ that works as immutable past snapshots of the configuration in a particular moment, with their description and release note, can be viewed and deleted through the versions management page:

Reviewer: ```suggestion
Versions works as immutable snapshots of the configuration in a particular moment, with their description and release note, can be viewed and deleted through the versions management page:
```

![Versions management page](img/versions-overview.png)

### Revision history

> To keep track of the Project evolution, the user can create `versions` (similar to Git tags) that act as immutable snapshots of the configuration in a particular moment. To keep track of how a revision has changed over time, the user can access its [revision history](#revision-history), which contains past snapshots of the Project configuration.

Author: I don't think it is necessary to link the #revision-history section here, since we are already inside it

Reviewer 2: Thanks for the catch. Probably a mistake after the merge from main.

> ![Revision History modal](img/revision-history.png)

Author: this screenshot shows a little bug on the last column, where buttons should be aligned to the right. Let's try to take a snapshot without the bug (it should suffice to reduce the browser width a little bit)

Reviewer 2: Coming.

> The revision history, accessible by clicking the related icon inside the branch selection area, contains a list of snapshots of the Project configuration, created each time it has been saved, on a certain revision or version. It also includes useful information such as the snapshot date, author, associated message, and any tags created with it. Moreover, the right-side button will show the differences between the selected snapshot and the previous one, in order to have a better understanding of the changes applied to the revision.

Author: ```suggestion
The revision history is accessible by clicking the related icon inside the branch selection area. It contains a list of snapshots of the Project configuration on the current revision or version, created each time it has been saved. It also includes useful information such as the snapshot date, author, associated message, and any tags associated with it. Moreover, the right-side button allows the user to visualize changes between the selected snapshot and the previous one, in order to better understand the configuration's evolution over time.
```

Author: I would add a snapshot of the diff editor

Reviewer 2: Done.

![Versions management page](img/versions-overview.png)

### Revision history

> To keep track of the Project evolution, the user can create `versions` (similar to Git tags) that act as immutable snapshots of the configuration in a particular moment. To keep track of how a revision has changed over time, the user can access its revision history, which contains past snapshots of the Project configuration.

Reviewer: We start the paragraph for revision history speaking about versions; that sounds odd.

I'd keep this paragraph in its previous location or rephrase it to emphasize revision history

```suggestion
To keep track of how a revision has changed over time, the user can access its revision history, which contains past snapshots of the Project configuration.
```

And put somewhere else the sentence 

```
To keep track of the Project evolution, the user can create `versions` (similar to Git tags) that act as immutable snapshots of the configuration in a particular moment. 
```

Reviewer 2: Description of `version` has been moved to the Version paragraph describing the Versions Management page. Let me know if it looks better.

Reviewer: sounds great. I've removed references to git tags.
